### PR TITLE
Add hive table function

### PR DESCRIPTION
**Pull request by lgbo-ustc**
Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add new table function `hive`, usage as follow
```
hive('<hive metastore url>', '<hive database>', '<hive table name>', '<columns definition>', '<partition columns>')
```
for example
```
SELECT * FROM hive('thrift://hivetest:9083', 'test', 'demo', '`id` Nullable(String), `score` Nullable(Int32), `day` Nullable(String)', 'day')
```

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/TableFunctions/Hive/TableFunctionHive.cpp
Comment (kssenii) [inline src/TableFunctions/Hive/TableFunctionHive.cpp:49]: args.size() is 5, why don't you start from zero index?

Comment (lgbo-ustc) [reply src/TableFunctions/Hive/TableFunctionHive.cpp]: This is my fault.

**Comment by lgbo-ustc** [comment]
There is another modification for improvement related to this, but rely on a new api at #34940.
Should I just wait for  #34940 be merged and commit this modification later in this pr ?

**Comment by lgbo-ustc** [comment]
The cmake failure seems not relate to the CmakeList.txt modification in this pr.  
```
cmake .. -DCMAKE_CXX_COMPILER=clang++-13 -DCMAKE_C_COMPILER=clang-13 -DENABLE_LIBRARIES=0 -DENABLE_TESTS=0 -DENABLE_UTILS=0 -DENABLE_EMBEDDED_COMPILER=0 -DENABLE_THINLTO=0 -DUSE_UNWIND=1 -DENABLE_NURAFT=1 -DENABLE_JEMALLOC=1 -DENABLE_REPLXX=1
```
I checkout to `master` branch, and meet the same issue.

**Comment by kssenii** [comment]
> Should I just wait for https://github.com/ClickHouse/ClickHouse/pull/34940 be merged and commit this modification later in this pr ?

as you wish, I can merge this pr right now as here checks passed already and you can add modifications for this pr in https://github.com/ClickHouse/ClickHouse/pull/34940 or we can wait for https://github.com/ClickHouse/ClickHouse/pull/34940 and merge this one after.

**Comment by lgbo-ustc** [comment]
> > Should I just wait for #34940 be merged and commit this modification later in this pr ?
> 
> as you wish, I can merge this pr right now as here checks passed already and you can add modifications for this pr in #34940 or we can wait for #34940 and merge this one after.

l will add the modification in #34940 , since there is no need to wait each other. 

It is about moving some initialization out of the constructor of StorageHive, which may cause some performance issue under the usage of `TableFunctionHive`. And run the initialization as need

**Review by kssenii — APPROVED** [review]

**Comment by yangshike** [comment]
when i used :
SELECT * FROM hive('thrift://xxx:9083', 'ods_qn', 'ods_crm_prod_on_off_line_msg_es_df', 'bizid Nullable(String), corpid Nullable(Int32),time Nullable(Int64),reasontype Nullable(Int32),weworkid Nullable(Int64), type Nullable(Int8),pt String', 'pt');

i get:

Received exception from server (version 22.3.2):
Code: 210. DB::Exception: Received from localhost:9000. DB::Exception: Unable to connect to HDFS: InvalidParameter: Cannot create namenode proxy,  does not contain host or port. (NETWORK_ERROR)

how can i do? thank you

**Comment by yangshike** [comment]
when i used :
SELECT * FROM hive('thrift://xxx:9083', 'ods_qn', 'ods_crm_prod_on_off_line_msg_es_df', 'bizid Nullable(String), corpid Nullable(Int32),time Nullable(Int64),reasontype Nullable(Int32),weworkid Nullable(Int64), type Nullable(Int8),pt String', 'pt');

i get:

Received exception from server (version 22.3.2):
Code: 210. DB::Exception: Received from localhost:9000. DB::Exception: Unable to connect to HDFS: InvalidParameter: Cannot create namenode proxy, does not contain host or port. (NETWORK_ERROR)

ps:
my hdfs used HA mode
this is my clickhouse config.xnl about hdfs:

<libhdfs3_conf>/etc/clickhouse-server/hdfs-client.xml</libhdfs3_conf>
how can i do? thank you

**Pull request merged**
